full_storage_utilization_test.py

```diff
@@ -0,0 +1,195 @@
+import time
+from sdcm.tester import ClusterTester
+
+
+class FullStorageUtilizationTest(ClusterTester):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.soft_limit = 75
+        self.hard_limit = 90
+        self.stress_cmd_30mins = self.params.get('stress_cmd')
+        self.add_node_cnt = self.params.get('add_node_cnt')
+        self.action_type = self.params.get('action_type')
+
+    def prepare_dataset_layout(self, dataset_size, row_size=10240):
+        n = dataset_size * 1024 * 1024 * 1024 // row_size
+        seq_end = n * 100
+
+        return f'cassandra-stress write cl=ONE n={n} -mode cql3 native -rate threads=10 -pop dist="uniform(1..{seq_end})" ' \
+               f'-col "size=FIXED({row_size}) n=FIXED(1)" -schema "replication(strategy=NetworkTopologyStrategy,replication_factor=3)"'
+
+    def setUp(self):
+        super().setUp()
+        self.start_time = time.time()
+
+    def start_slow_write(self):
+        self.run_stress_thread(stress_cmd=self.stress_cmd_30mins, keyspace_name=f"{self.action_type}")
+        self.log.info("Wait for 2 mins for the stress command to start")
+        time.sleep(120)        
```

Comment:
consider adding some constant definition variables for the sleep values

Reply:
ok, will update. waited 2 mins, so that throttled write operation will be running while we perform either scale-out or scale-in. Basic idea to have on-going read/write during topology change, its one of the suggestion by @lukepio

Reply:
general recommendation

if you put sleep in test code, even if it's configurable, it should have a clear reasoning in a comment before it.

regardless we should always look for a way to way for the actual "thing" we are really waiting for

in this case you probably wanted to wait for writes to start, and we have such example in the rolling upgrade tests:

```
self.metric_has_data(
            metric_query='sct_cassandra_stress_write_gauge{type="ops", keyspace="keyspace1"}', n=5)
```

it's a way to wait for a specific metric to appear

it might not fit perfectly, but you might be able to extend to fit your needs.